src/components/LogView/index.js

```diff
@@ -366,7 +369,47 @@ class LogView extends React.Component<Props, State> {
     }
   }
 
+  componentDidMount() {
+    if (this.props.wrap) {
+      this.props.toggleWrap();
+    }
+  }
+
   componentDidUpdate(prevProps: Props, prevState: State) {
+    const currBookmarksSet = new Set(
+      this.props.bookmarks.map(({ lineNumber }) => lineNumber)
+    );
+    const lastLineNo = this.state.lines.length - 1;
+    // handle initial bookmark scroll
+    if (
+      // Don't scroll unless lines are rendered and bookmarks exist
+      this.state.lines.length > 0 &&
+      prevState.lines.length === 0 &&
+      this.props.bookmarks.length &&
+      // Don't scroll when the bookmarks look like [0, last line]
+      !(
+        currBookmarksSet.size === 2 &&
+        currBookmarksSet.has(0) &&
```

Comment:
Is this check necessary? I wonder if you can just always scroll to the first line after the log is loaded, because if it's the default line 0 then you're at the top anyway

Reply:
I removed line 392 and noticed scroll stuttering on load. I think we should leave this check in since it describes a default state when no bookmarks are explicitly provided by user